src/components/Item/Item.component.jsx

```diff
@@ -9,18 +10,45 @@ const Item = props => {
   const itemId = props.id.toString();
   const localToken = props.localToken;
   const over24 = props.over24;
+  let lastEstimate = props.lastEstimate;
+  let latestInterval = props.latestInterval;
+  let numberOfPurchases = props.numberOfPurchases;
+  let nextPurchaseInterval = props.nextPurchaseInterval;
+  let lastPurchaseDate = props.date;
 
   useEffect(() => {
     if (over24 === false) {
       document.getElementById(itemId).setAttribute('class', 'item__hightlight');
     }
   });
 
-  //To update the purchase date
+  // To update the purchase date
   const markPurchased = event => {
     event.preventDefault();
     const date = new Date();
 
+    // To update the number of purchases
+    numberOfPurchases += 1;
+
+    // To update the latest interval
+    if (lastPurchaseDate === null) {
+      latestInterval = 0;
+    } else {
+      let currentTimeInSeconds = new Date().getTime() / 1000;
+      let lastPurchasedTimeInSeconds = lastPurchaseDate.seconds;
+      latestInterval = Math.ceil(
+        (currentTimeInSeconds - lastPurchasedTimeInSeconds) / 86400,
```

Comment:
This `86400` is used in multiple places. If you wanted to, you could have something like
`const SECONDS_IN_DAYS = 86400` that you could use. This would also allow you to remove the comments about how many seconds are in the day.

Reply:
This is a great suggestion! It avoids the use of "magic numbers." You can read more about them here:
https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

Reply:
Thanks for the helpful suggestions, Joey! Thanks for the reference link, Jen!